net/trojan/Portfile

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+github.setup        trojan-gfw trojan 1.15.1 v
+revision            0
+categories          net
+maintainers         {@i0ntempest me.com:szf1234} openmaintainer
+platforms           darwin
+license             GPL-3
+
+description         An unidentifiable mechanism that helps you bypass network restrictions.
+long_description    Trojan features multiple protocols over TLS to avoid both \
+                    active/passive detections and ISP QoS limitations. It doesn't rely on \
+                    strong encryption or obfuscation\; instead it imitates the most \
+                    common protocol, HTTPS, so firewalls won't see anything other than \
+                    normal HTTPS traffic.
+
+checksums           rmd160  b36ac226c6d1cd56c37cd1ef675e0c11fb2d5f19 \
+                    sha256  8627a7cc6e1ebaca93a93cb19e4cc8aac853e1602f117c0470e43db916fded68 \
+                    size    48139
+
+depends_lib-append  port:boost \
+                    path:lib/libssl.dylib:openssl
+
+configure.args-append \
+                    -DCMAKE_INSTALL_SYSCONFDIR=${prefix}/etc \
+                    -DDEFAULT_CONFIG=${prefix}/etc/trojan/config.json \
+                    -DENABLE_MYSQL=OFF
+
+pre-configure {
+    reinplace "s|/etc/trojan/config.json|${prefix}/etc/trojan/config.json|g" \
+        ${worksrcpath}/docs/trojan.1
+}
+
+post-destroot {
+    move ${destroot}${prefix}/etc/trojan/config.json ${destroot}${prefix}/etc/trojan/config.json.example
+}
+
+post-activate {
+    if {![file exists ${prefix}/etc/trojan/config.json]} {
+        copy ${prefix}/etc/trojan/config.json.example ${prefix}/etc/trojan/config.json
+    }
+}
+
+notes-append "
+    The default config file is copied to ${prefix}/etc/trojan/config.json and it's content\
+    will be preserved across upgrades and reinstalls. The startup item will use this config\
+    file by default.
+"
+
+startupitem.create          yes
+startupitem.executable      ${prefix}/bin/trojan --config=${prefix}/etc/trojan/config.json
+
+proc dbvariants {dbport prefix} {
```

Comment:
you can avoid having to pass `prefix` in in each call by just adding the global variable to the scope of the proc. i.e. just add

```
global prefix
```

and then just remove it from the arguments...

Reply:
Thanks